client/basic-authorship/src/basic_authorship.rs

```diff
@@ -232,7 +232,7 @@ impl<A, B, Block, C> ProposerInner<B, Block, C, A>
 		}
 
 		// proceed with transactions
-		let block_timer = self.metrics.report(|metrics| metrics.block_constructed.start_timer());
```

Comment:
If the whole function will return because of error, this will be just dropped and time recorded in some uncontrollable manner